app/lib/features/public_room_search/widgets/public_room_search.dart

```diff
@@ -0,0 +1,332 @@
+    return DropdownMenu<FilterBy>(
+      initialSelection: selected,
+      // requestFocusOnTap is enabled/disabled by platforms when it is null.
+      // On mobile platforms, this is false by default. Setting this to true will
+      // trigger focus request on the text field and virtual keyboard will appear
+      // afterward. On desktop platforms however, this defaults to true.
+      requestFocusOnTap: true,
+      enableSearch: false,
+      label: const Text('Show only'),
+      textStyle: TextStyle(color: Theme.of(context).hintColor),
+      menuStyle: const MenuStyle(visualDensity: VisualDensity.compact),
+      onSelected: (FilterBy? newFilter) {
+        ref
+            .read(searchFilterProvider.notifier)
+            .updateFilters(newFilter ?? FilterBy.both);
+      },
+      dropdownMenuEntries: const [
+        DropdownMenuEntry<FilterBy>(
+          value: FilterBy.both,
+          label: 'All',
+        ),
+        DropdownMenuEntry<FilterBy>(
+          value: FilterBy.spaces,
+          label: 'Spaces',
+        ),
+        DropdownMenuEntry<FilterBy>(
+          value: FilterBy.chats,
+          label: 'Chats',
+        ),
+      ],
+    );
+  }
+
+  Widget _searchResults(BuildContext context, WidgetRef ref) {
+    return RiverPagedBuilder<Next?, PublicSearchResultItem>.autoDispose(
+      firstPageKey: const Next(isStart: true),
+      provider: publicSearchProvider,
+      // pullToRefresh: true,
+      firstPageProgressIndicatorBuilder: (context, controller) =>
+          loadingPage(context),
+      itemBuilder: (context, item, index) => PublicRoomItem(
+        item: item,
+        onSelected: (item) =>
+            widget.onSelected(item, ref.read(searchFilterProvider).server),
+      ),
+      noItemsFoundIndicatorBuilder: (context, controller) {
+        if (ref.read(publicSearchProvider.notifier).isLoading()) {
+          return loadingPage(context);
+        }
+        return nothingFound(context);
+      },
+      pagedBuilder: (controller, builder) => PagedSliverList(
+        pagingController: controller,
+        builderDelegate: builder,
+      ),
+    );
+  }
+
+  Widget nothingFound(BuildContext context) {
+    return Center(
+      child: Text(L10n.of(context).searchPublicDirectoryNothingFound),
+    );
+  }
+
+  Widget loadingPage(BuildContext context) {
+    return Column(
+      children: List.generate(
+        5,
+        (index) => Card(
+          child: Column(
+            mainAxisSize: MainAxisSize.min,
+            mainAxisAlignment: MainAxisAlignment.start,
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: [
+              Flexible(
+                child: Padding(
+                  padding: const EdgeInsets.symmetric(vertical: 5),
+                  child: ListTile(
+                    leading: const Bone.square(size: 30),
+                    title: Skeletonizer(
+                      child: Text(
+                        'nothing',
```

Comment:
Loading texts should also be formatted as l10n strings...

Reply:
no... why? this is in a skeletonizer, it is never shown.... in the latest skeletonizer we finally have the `Bone` class that will allow us to not have these pointless filler anymore... \o/

Reply:
sounds plausible if there's no way of it showing up. 👍🏻